custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -246,6 +247,7 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
     };
 
     vm.onSelectLocation = function ($item, level) {
+        vm.selectedLocationLevel = level + 1;
```

Comment:
can you explain this?

Reply:
This is to achieve the below functionality. Which is that selectedLevel which is updated in `locationsService.onSelectLocation` is being updated by both location filter and viewBy filter.
So this variable is to keep the selectedlevel track based on location filter.

Reply:
Got it. I think that probably warrants a comment as it's rather confusing.

And confirming there's no way to just always pull it from the locationsService so we don't have to deal with maintaining this state in two places?

Reply:
Hey, changed the way to use it. I am finding same thing via selectedLocation list size. I found a function for that. So just reusing that. I think its better to reuse that